espree.js

```diff
@@ -3356,10 +3356,15 @@ function reinterpretAsCoverFormalsList(expressions) {
 
 function parseArrowFunctionExpression(options, marker) {
     var previousStrict, body;
+    var arrowStart = lineNumber;
 
     expect("=>");
     previousStrict = strict;
 
+    if (lineNumber > arrowStart) {
+        throwError({}, Messages.UnexpectedToken, "=>");
```

Comment:
In this case the `=>` is on a new line by itself and so it's not expected, at least not there. Let me know if that doesn't sound right.

```
var a = ()
	=> 0; 
```